### ISSUE TITLE
cmssw-el8 fails on vocms006 and vocms007

### ISSUE THREAD

**Issue by Martin-Grunewald** [cms-sw/cmssw#39667]
The error message is:

```
gruen@vocms007:~/[102]$ cmssw-el8
WARNING: DEPRECATED USAGE: Environment variable SINGULARITY_BINDPATH will not be supported in the future, use APPTAINER_BINDPATH instead
ERROR  : Failed to create user namespace: user namespace disabled
```

While the warning could perhaps be ignored for now, it seems the feature of ```user namespace```s has gotten lost within the last few days. How can it be restored? There should be a way given that it works on lxplus7 (neither WARNING nor ERROR), no?

**Comment by fwyzard** [comment]
I think I "fixed" `vocms007`.
Can you give it a try ?

I need to reboot `vocms006` in order to apply the same changes - let me know when I can do that.

**Comment by Martin-Grunewald** [comment]
Great, thanks, on vocms007 the error is gone now, the WARNING is still there, and the shell is in
the singularity prompt:

```
gruen@vocms007:~/[101]$ cmssw-el8
WARNING: DEPRECATED USAGE: Environment variable SINGULARITY_BINDPATH will not be supported in the future, use APPTAINER_BINDPATH instead
Singularity> 
```
Feel free to reboot vocms006 asap...

**Comment by fwyzard** [comment]
For the record:
  - add `namespace.unpriv_enable=1 user_namespace.enable=1` to `GRUB_CMDLINE_LINUX=...` in `/etc/default/grub`
  - regenerate `grub.conf` with `sudo grub2-mkconfig -o /boot/efi/EFI/centos/grub.cfg`
  - add `user.max_user_namespaces = 15000` to `/etc/sysctl.conf`
  - reboot

**Comment by fwyzard** [comment]
Done.

**Comment by Martin-Grunewald** [comment]
Thanks a lot @fwyzard !
What is left now is the above WARNING, and that seems to be specific to the cmssw-el8 script.... 
@smuzaffar

**Comment by aandvalenzuela** [comment]
Hi @Martin-Grunewald, I think this warning is related to the fact that Singularity has been "recently" rebranded to Apptainer. I will raise this concert to @smuzaffar on Monday since I guess we will need to adapt our container to Apptainer now.

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/cmssw-osenv/pull/8 should fix the `WARNING: DEPRECATED USAGE: Environment variable SINGULARITY_BINDPATH will not be supported in the future, use APPTAINER_BINDPATH instead` warning

new `cmssw-*` commands now prefer to use `apptainer` if available ( e.g. on `cs8` systems) otherwise if falls based to use `singularity` (e.g. CentOS7 systems)

**Comment by Martin-Grunewald** [comment]
@smuzaffar 
Thank you!

**Comment by smuzaffar** [comment]
@fwyzard , for the record, we also saw these `Failed to create user namespace: user namespace disabled` error after the `singularity` to `apptainer` updated. The fix was to installed `apptainer-suid` package too ( which I think does the same what you did in https://github.com/cms-sw/cmssw/issues/39667#issuecomment-1271208292 ) but no reboot was needed.

**Comment by fwyzard** [comment]
No, it is definitely not the same thing - and we should **not** run singularity/apptainer suid!

I would strongly suggest to revert the installation of `apptainer-suid`, and instead enable the user namespaces in the kernel.

See https://apptainer.org/docs/user/main/security.html for more information.

**Comment by smuzaffar** [comment]
I can not revert `apptainer-suid` as it was installed by CERN IT team on our cvmfs publisher nodes
